packages/emotion/src/InstUISettingsProvider/index.tsx

```diff
@@ -27,14 +27,26 @@ import { TextDirectionContext } from '@instructure/ui-i18n'
 import { ThemeProvider } from '@emotion/react'
 import { ThemeOrOverride } from '../EmotionTypes'
 import { getTheme } from '../EmotionThemeProvider'
-import { DeterministicIdContextProvider } from '@instructure/ui-react-utils'
+import {
+  DeterministicIdContextProvider,
+  getElementType
+} from '@instructure/ui-react-utils'
 import type { DeterministicIdProviderValue } from '@instructure/ui-react-utils'
+import type { AsElementType } from '@instructure/shared-types'
 
 type InstUIProviderProps = {
   theme?: ThemeOrOverride
-  dir?: 'ltr' | 'rtl' // TODO allow "auto" too
   instanceCounterMap?: DeterministicIdProviderValue
-}
+} & (
```

Comment:
For me TS doesn't underline if I add the `as` prop without the `dir` :(

Reply:
Should be fixed:
<img width="1167" alt="Screenshot 2022-03-24 at 10 23 48" src="https://user-images.githubusercontent.com/20640237/159893397-7d54116c-6816-4b87-b286-1f2c3f7cdcde.png">